### DISCUSSION TITLE
FileNotFoundError for madmax 'chia_plot'

### DISCUSSION THREAD

**Discussion by tcwatson** [ericaltendorf/plotman#838] [Q&A]
This is probably user error, but wanted to report this just in case. 

When configuring plotman 0.5 to use madmax, it's unable to find the command 'chia_plot'. When I was using madmax by itself, I was referencing it via its fully qualified path, so at first I thought an alias might fix it, but it did not. Eventually I ended up modifying manager.py to fully qualify the path to the chia_plot command, which did finally work. 

``` shell
(venv) tyler@chiaplotter:~/chia-blockchain$ plotman interactive
Traceback (most recent call last):
  File "/home/tyler/chia-blockchain/venv/bin/plotman", line 8, in <module>
    sys.exit(main())
  File "/home/tyler/chia-blockchain/venv/lib/python3.8/site-packages/plotman/plotman.py", line 243, in main
    interactive.run_interactive(
  File "/home/tyler/chia-blockchain/venv/lib/python3.8/site-packages/plotman/interactive.py", line 352, in run_interactive
    curses.wrapper(
  File "/usr/lib/python3.8/curses/__init__.py", line 105, in wrapper
    return func(stdscr, *args, **kwds)
  File "/home/tyler/chia-blockchain/venv/lib/python3.8/site-packages/plotman/interactive.py", line 129, in curses_main
    (started, msg) = manager.maybe_start_new_plot(
  File "/home/tyler/chia-blockchain/venv/lib/python3.8/site-packages/plotman/manager.py", line 227, in maybe_start_new_plot
    p = subprocess.Popen(plot_args,
  File "/usr/lib/python3.8/subprocess.py", line 858, in __init__
Traceback (most recent call last):
  File "/home/tyler/chia-blockchain/venv/bin/plotman", line 8, in <module>
    sys.exit(main())
  File "/home/tyler/chia-blockchain/venv/lib/python3.8/site-packages/plotman/plotman.py", line 243, in main
    interactive.run_interactive(
  File "/home/tyler/chia-blockchain/venv/lib/python3.8/site-packages/plotman/interactive.py", line 352, in run_interactive
    curses.wrapper(
  File "/usr/lib/python3.8/curses/__init__.py", line 105, in wrapper
    return func(stdscr, *args, **kwds)
  File "/home/tyler/chia-blockchain/venv/lib/python3.8/site-packages/plotman/interactive.py", line 129, in curses_main
    (started, msg) = manager.maybe_start_new_plot(
  File "/home/tyler/chia-blockchain/venv/lib/python3.8/site-packages/plotman/manager.py", line 227, in maybe_start_new_plot
    p = subprocess.Popen(plot_args,
  File "/usr/lib/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.8/subprocess.py", line 1704, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'chia_plot'
```

Modified line in manager.py that works: 
``` diff
if plotting_cfg.type == "madmax":
                if plotting_cfg.madmax is None:
                    raise Exception(
                        "madmax plotter selected but not configured, report this as a plotman bug",
                    )
-                 plot_args = ['chia_plot',
+                 plot_args = ['/home/tyler/madmax/chia-plotter/build/chia_plot',
                    '-n', str(1),
                    '-r', str(plotting_cfg.madmax.n_threads),
                    '-u', str(plotting_cfg.madmax.n_buckets),
                    '-t', tmpdir if tmpdir.endswith('/') else (tmpdir + '/'),
                    '-d', dstdir if dstdir.endswith('/') else (dstdir + '/') ]
                if dir_cfg.tmp2 is not None:
                    plot_args.append('-2')
                    plot_args.append(dir_cfg.tmp2 if dir_cfg.tmp2.endswith('/') else (dir_cfg.tmp2 + '/'))
```

Not sure if this is relevant, but I updated both plotman and madmax to their latest versions today after being about 2 months out of date on both. 

Is there some other method of telling linux (other than an alias) or even python that `chia_plot` should refer to a specific path? Am I missing something with setting up madmax? Or should plotman be configurable to set a custom path to chia_plot? 

Thanks!

**Comment by altendky** [comment]
The "quick" way would be to run `PATH=/home/tyler/madmax/chia-plotter/build:$PATH plotman interactive`.  The better way would be to add that to your path permanently in your `~/.profile`.  You have to logout and back in to get the change in effect everywhere.  plotman will likely also get the ability to specify the path explicitly in the configuration file (https://github.com/ericaltendorf/plotman/pull/823).

**Reply by TaylorM87** [reply]
I cannot for the life of me find how to add this path to profile and get it to work. I have added to /etc/environment as well as added another entry to ~/.profile. Do you have a sample entry for this for profile? I logged out and back in as well.

**Reply by altendky** [reply]
Maybe share what you added.

**Reply by TaylorM87** [reply]
Sorry that was horrible request for help. Thanks for following up, I have been using the following to launch it until I figure out the profile deal...PATH=/home/taylor/chia-plotter/build:$PATH plotman interactive
Added the following to /etc/environment ":/usr/chia-plotter/build"
Added to ~/.profile (I also added the full path /home/taylor/chia-plotter/build below as well)
if [ -d "$HOME/chia-plotter/build" ] ; then
    PATH="$HOME/chia-plotter/build:$PATH"
fi

**Reply by altendky** [reply]
What do you get for `echo $PATH` and `ls -l /home/taylor/chia-plotter/build/chia_plot`?

**Reply by TaylorM87** [reply]
/sbin:/bin:/usr/bin:/usr/local/bin:/snap/bin
&
/home/taylor/chia-plotter/build/chia_plot
respectively

**Reply by altendky** [reply]
Your path modifications don't seem to be sticking.  Might be worth finding some interactive Linux help to walk through and check everything.

**Comment by tcwatson** [comment]
Oh right, PATH! Thanks so much, worked like a charm. 

For anyone who finds this later, I'm not sure if it mattered or not, but I noticed all the existing PATH additions in my .profile were wrapped in quotes, like `PATH="/home/tyler/madmax/chia-plotter/build:$PATH"` so I added it like that at the bottom of the file.
